app/server/src/features/accounts/methods.ts

```diff
@@ -209,4 +211,57 @@ export async function deleteAccount(prisma: PrismaClient, input: DeleteAccountFo
 
     // return deleted user
     return { deletedUser };
-}
\ No newline at end of file
+}
+
+export async function resetPasswordRequest(prisma: PrismaClient, input: ResetPasswordRequestForm) {
+    const result = await prisma.user.findUnique({
+        where: { email: input.email }
+    });
+
+    const accountFound = !!result;
+
+    // No need to send email if the account does not exist
+    if (!accountFound) throw new Error('No account with that email was found');
+
+    const token = await jwt.sign({ email: input.email });
+    const passwordResetLink = `prytaneum.io/reset-password?token=${token}`
+
+    try {
+        await sendEmail({
+            to: input.email,
+            subject: 'Password Reset',
+            template: 'password-reset',
+            'h:X-Mailgun-Variables': JSON.stringify({ passwordResetLink }),
+        });
+        return accountFound;
+    } catch (err) {
+        server.log.error(err);
+        throw new Error(errors.email)
+    }
+}
+
+export async function resetPassword(prisma: PrismaClient, input: ResetPasswordForm) {
+    const { newPassword, confirmNewPassword, token } = input;
+    let email;
+
+    // Ensure token is valid
+    try {
+        const result = await jwt.verify(token) as { email?: string };
+        if (!result.email) throw new Error(errors.jwt);
+        email = result.email;
+    } catch (err) {
+        server.log.error(err);
+        throw new Error(errors.jwt);
+    }
+
+    // Ensure passwords match
+    if (newPassword !== confirmNewPassword) throw new Error('Passwords must match');
+
+    const encryptedPassword = newPassword ? await bcrypt.hash(newPassword, 10) : null;
```

Comment:
Do we ensure the new password is non-empty?  Also, don't we have a generic hash password function with all the appropriate settings already passed in?  Would we ever want the password to be null?